polynote-kernel/src/main/scala/polynote/kernel/lang/scal/ScalaSource.scala

```diff
@@ -223,7 +242,9 @@ class ScalaSource[G <: Global](
       val wrappedSource = atPos(range) {
         global.PackageDef(
           atPos(beginning)(global.Ident(global.TermName(notebookPackage))),
-          directImports.map(forcePos(beginning, _)).map(global.resetAttrs) ::: List(
+          directImports.map(forcePos(beginning, _)).map(global.resetAttrs) :::
+          liftedTrees.map(forcePos(beginning, _)) ::: // TODO: this probably messes up the position settings, hopefully we can do something about that...
```

Comment:
If this tree-lifting were done in `compileUnitWithSubstitutions` then positioning would be fine. Any reason to do it here?

Reply:
Maybe we should try just turning off position validation. Then we could keep the lifted trees' original positions even though they wouldn't "make sense" anymore.

Reply:
I can play around with it and see what it looks like. I think it won't solve all our positioning problems though, because ResultValues will still take positions from the code that's ultimately compiled which will have the mangled positions. 

The reason to do it here was just because it was a bit easier.